just_audio_background/lib/just_audio_background.dart

```diff
@@ -839,3 +859,42 @@ extension _ValueStreamExtension<T> on ValueStream<T> {
   /// Backwards compatible version of valueOrNull.
   T? get nvalue => hasValue ? value : null;
 }
+
+/// Notification config, used to set system notification properties.
+@immutable
+class NotificationConfig {
+  const NotificationConfig({
+    this.controls = const [],
+    this.systemActions = const {},
+    this.androidCompactActionIndices,
+  });
+
+  /// Not supported starting with Android 13 (API level 33)
+  /// 
+  /// More info: https://developer.android.com/about/versions/13/behavior-changes-13#playback-controls
+  final List<int>? androidCompactActionIndices;
+  final List<MediaControl> controls;
+  final Set<MediaAction> systemActions;
+}
+
+
+/// Snapshot of current player state.
+/// Used to provide data to [NotificationConfigBuilder].
+abstract class PlayerStateSnapshot {
```

Comment:
It's unfortunate to replicate existing state classes that are in just_audio, but just_audio_background should not depend on just_audio so this is understandable for now.

Eventually, I think these state classes should be defined in just_audio_platform_interface where they can be reused.

Reply:
So, is this in scope of this PR, if so, tell me where I should move it, please?

Reply:
That comment was just to recognise the need for something now even though it should be moved elsewhere in the future.

But as it is, is  this really a "snapshot"? It looks like you're using this as a mixin (although you defined it as an abstract class instead of a mixin). That means that this package is now returning a reference to an internal class which it probably shouldn't. If the so-called snapshot is actually an object where the values may change behind the scenes, then it is not really a snapshot. I think it would be far less surprising to create an immutable state object, and to build a new instance of it each time a new state needs to be broadcast.

As for naming, and considering where these state objects will likely move in the future, I would say we do not know at this stage what that future state object will look like, and anything we might come up with now would be rather narrow in its vision. So to avoid any future problems, it might be best to choose a class name for this state that is obviously specific to this particular package rather than some name that we might want to reserve for the future ideal API.